.github/scripts/check_pr_quality.py

```diff
@@ -1,112 +1,261 @@
 import os
 import re
 import sys
+from enum import Enum
 from github import Github
+from dataclasses import dataclass
+from typing import List
+
+
+class IssueType(Enum):
+    TUTORIAL = "tutorial"
+    MISSING_TICKET = "missing_ticket"
+    DJANGO_TUTORIAL = "django_tutorial"
+
+
+class Severity(Enum):
+    HIGH = "high"
+    MEDIUM = "medium"
+    LOW = "low"
+
+
+class Location(Enum):
+    TITLE = "title"
+    BODY = "body"
+
+
+class Confidence(Enum):
+    LOW = "low"
+    MEDIUM = "medium"
+    HIGH = "high"
+
+
+@dataclass
+class PatternMatch:
+    """Stores information about a matched pattern"""
+
+    pattern: str
+    matched_text: str
+    location: Location
+    context: str  # surrounding text
+
+
+@dataclass
+class QualityIssue:
+    """Represents a quality issue found in a PR"""
+
+    type: IssueType
+    severity: Severity
+    message: str
+    matches: List[PatternMatch]
+
+    def should_auto_close(self) -> bool:
+        """Determine if this issue warrants auto-closing"""
+        if self.type != IssueType.TUTORIAL:
```

Comment:
Like `None`, it's idiomatic to compare enums with `is` and `is not`:

```suggestion
        if self.type is not IssueType.TUTORIAL:
```